query-node/mappings/src/content/app.ts

```diff
@@ -1,43 +1,46 @@
 import { DatabaseManager, SubstrateEvent } from '@joystream/hydra-common'
-import { ICreateApp, IUpdateApp } from '@joystream/metadata-protobuf'
+import { ICreateApp, IDeleteApp, IUpdateApp } from '@joystream/metadata-protobuf'
 import { integrateMeta } from '@joystream/metadata-protobuf/utils'
-import { ChannelId } from '@joystream/types/primitives'
-import { logger } from '@joystream/warthog'
-import { Channel, App } from 'query-node/dist/model'
-import { inconsistentState } from '../common'
+import { App, Membership } from 'query-node/dist/model'
+import { logger, inconsistentState } from '../common'
 
 export async function processCreateAppMessage(
   store: DatabaseManager,
   event: SubstrateEvent,
-  channelId: ChannelId,
-  message: ICreateApp
+  message: ICreateApp,
+  memberId?: string
 ): Promise<void> {
   const { name, appMetadata } = message
-  const appId = await createAppId(event)
+  const appId = `${event.blockNumber}-${event.indexInBlock}`
+
+  let ownerMember: Membership | undefined
+  if (memberId) {
+    ownerMember = await store.get(Membership, {
+      where: {
+        id: memberId,
+      },
+    })
+    if (!ownerMember) {
+      return inconsistentState(`Member with ${memberId} not found`)
+    }
+  }
 
   const isAppExists = await store.get(App, {
     where: {
-      name: name,
+      name: message?.name,
```

Comment:
message cannot be empty, so it should be either:
```suggestion
      name: message.name,
```
or `message` should have `ICreateApp | undefined` type (if it's possible for it to be empty).